python/ql/lib/semmle/python/dataflow/new/internal/LocalSources.qll

```diff
@@ -225,4 +236,26 @@ private module Cached {
       n = call.getFunction()
     )
   }
+
+  /**
+   * Holds if `node` flows to a value that, when awaited, results in `awaited`.
+   */
+  cached
+  predicate await(LocalSourceNode node, Node awaited) {
+    exists(Node awaitedValue |
+      node.flowsTo(awaitedValue) and
+      awaited = awaited(awaitedValue)
+    )
+  }
+
+  /**
+   * Holds if `node` flows to a sequence of which `subscript` is a subscript.
```

Comment:
NIT: "Sequence" is a slightly wrong term here. If we look at the [abstract base classes of Python](https://docs.python.org/3/library/collections.abc.html), a `dict` will not be considered a sequence, but can certainly be subscripted.

According to [the docs of `__getitem__`](https://docs.python.org/3/reference/datamodel.html#object.__getitem__), it seems like saying sequence/mapping will be good enough.

```suggestion
   * Holds if `node` flows to a sequence/mapping of which `subscript` is a subscript.
```